creator-node/src/services/sync/primarySyncFromSecondary.js

```diff
@@ -53,13 +53,21 @@ module.exports = async function primarySyncFromSecondary({
     )
 
     // TODO should be able to pass this through from StateMachine / caller
-    const userReplicaSet = await getUserReplicaSet({
+    let userReplicaSet = await getUserReplicaSet({
       wallet,
       selfEndpoint,
       logger,
       libs
     })
 
+    // Error if this node is not primary for user
+    if (userReplicaSet[0] !== selfEndpoint) {
+      throw new Error(`Failure - this node is not primary for user`)
+    }
```

Comment:
😬 this should have been locked down earlier

Reply:
check out my comment below and lmk what you think. we should fetch this data from chain vs discovery